docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,449 @@
+serialization of mutiple objects, such as model, maps, flux-points etc. the best
+option is likely to introduce the serialization with a YAML based index file:
+
+.. code ::
+
+    dataset = MapDataset.read("dataset.yaml") # lazy loading?
+    dataset.write("dataset.yaml")
+
+Where the index file points to the various files needed for initialization of the
+dataset. Here is an example:
+
+.. code::
+
+    dataset:
+        type: map-dataset
+        counts: "obs-123/counts.fits"
+        exposure: "obs-123/exposure.fits"
+        edisp: "obs-123/edisp.fits"
+        psf: "obs-123/psf.fits"
+        model: "model.yaml"
+        background-model: "obs-123/background.fits"
+
+Addtionally one could introduce a single FITS file serializiaton for quickly writing /
+reading datasets to disk.
+
+The `Datasets` object could be serialized equivalently as a list of datasets:
+
+.. code::
+
+    - dataset:
+        type: spectrum-dataset
+        maps:
+            counts: "obs-123/counts.fits"
+            exposure: "obs-123/exposure.fits"
+            edisp: "obs-123/edisp.fits"
+        model: "model.yaml"
+        background-model: "obs-123/background.fits"
+        likelihood: "wstat"
+
+    - dataset:
+        type: spectrum-dataset
+        maps:
+            counts: "obs-124/counts.fits"
+            exposure: "obs-124/exposure.fits"
+            edisp: "obs-124/edisp.fits"
+        model: "model.yaml"
+        background-model: "obs-124/background.fits"
+        likelihood: "wstat"
+
+    - dataset:
+        type: flux-point-dataset
+        flux-points : "fermipy-flux-points.fits"
+        model: "spectral-model.yaml"
+        likelihood: "template"
+
+
+Lazy loading of Datasets
+------------------------
+For the use-case of inspecting individual datasets (e.g. MapDataset per observation)
+it could be advantegeous to implement a lazy-loading or generator interface for
+`Datasets.read()`. Such that the individual datasets are only read from disk on
+requests and are not loaded in memory when calling `.read()`. We propose to handle
+this with a generic lazy-loading mechanism for `Map` objects, where the data is loaded
+on access of the `.data` attribute.
+
+
+Simulation of `MapDataset`
+--------------------------
+The `MapDataset` class will be the central class for map based analyses. In many
+cases it's useful to simulate counts from a predicted number of counts model. Two
+approaches are possible: simulating a counts map by calling `np.random.poisson()` on
+a predicted number of counts map or sampling of event lists. The latter is addressed
+in a separate PIG (#009). The direct sampling of binned data could be supported
+as following:
+
+.. code::
+
+    # allow counts=None on init
+    dataset = MapDataset(counts=None)
+
+    # to sample a counts map from npred
+    dataset.counts = dataset.simulate_counts(random_seed=0)
+
+    fit = Fit(dataset)
+    fit.optmimize()
+
+
+List of Pull Requests
```

Comment:
Replace the mention of milestone with links to issues / PRs?

Reply:
Sounds like a good idea, but it also duplicates the information from our changelog...I'll think about it once more.

Reply:
Python PEPs or Astropy APEs do refer to relevant PRs, I would suggest to do that.
But if you don't want to, then probably removing the milestone info here completely (since it's partially incorrect) is the simplest solution.

Reply:
I added the links to the corresponding PRs, if available.